FrostySdk/IO/EbxReaderRiff.cs

```diff
@@ -483,12 +476,33 @@ internal override TypeRef ReadTypeRef()
         {
             uint type = ReadUInt();
             Position += 4;
-            if ((type & 0x80000000) == 0)
+            if (type == 0)
             {
                 return new TypeRef("0");
             }
-            EbxFieldType valuetype = (EbxFieldType)((type >> 5) & 0x1F);
-            return new TypeRef(valuetype.ToString());
+
+            // (always?) primitive type
+            if ((type & 0x80000000) != 0)
+            {
+                type &= ~0x80000000;
+                EbxFieldType valuetype = (EbxFieldType)((type >> 5) & 0x1F);
+                return new TypeRef(valuetype.ToString());
+            }
+            else
+            {
+                int classRef = (int)(type >> 2);
+                int unkVal = (int)(type & 3);
+                Guid? classGuid = classGuids[classRef];
+                if (classGuid.HasValue)
+                {
+                    Type refClassType = TypeLibrary.GetType(classGuid.Value);
```

Comment:
It would exclude TypeRefs that reference a signed GUID, but I guess I can just have the `TypeRef` constructor fall back to `TypeLibrary.GetType` in case it can't find a type